### PR TITLE
MGMT-10281: Add documentation for the converged flow

### PR DESCRIPTION
**Pull request by eranco74**
Add documentation for the converged flow

- Should this PR be tested by the reviewer? no
- Is this PR relying on CI for an e2e test run? no
- Should this PR be tested in a specific environment? no
- Any logs, screenshots, etc that can help with the review process? no

## List all the issues related to this PR
https://issues.redhat.com/browse/MGMT-10281

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @flaper87
/cc @avishayt 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/hive-integration/README.md [hunk 1]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
...
 - Reconcile the BareMetalHost hardware details by copying the Agent's inventory data to the BMH's `hardwaredetails` annotation.
+- Disable ironic inspection disabled
 
 
 See BMAC documentation [here](./baremetal-agent-controller.md).
 
+## Bare Metal Operator Integration using the converged flow
+See reference [here](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/ztp-metal3.md)
+
+In case the Bare Metal Operator on the HUB cluster version is 4.11 or higher, the integration between the MBO and the assisted-service is slightly different.
```
Comment (romfreiman) [inline docs/hive-integration/README.md:173]: bmo?

File: docs/hive-integration/README.md [hunk 2]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
     - Hostname (optional for user to set)
     - MachineConfigPool (optional for user to set)
 - Reconcile the BareMetalHost hardware details by copying the Agent's inventory data to the BMH's `hardwaredetails` annotation.
+- Disable ironic inspection disabled
```
Comment (avishayt) [inline docs/hive-integration/README.md:165]: Failed to parse - did you mean:
```suggestion
- Disable ironic inspection
```

File: docs/hive-integration/README.md (hunk 1 above)
Comment (eranco74) [reply docs/hive-integration/README.md:173]: Bare Metal Operator AKA: BMO

File: docs/hive-integration/README.md [hunk 3]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
...
 
 
 See BMAC documentation [here](./baremetal-agent-controller.md).
 
+## Bare Metal Operator Integration using the converged flow
+See reference [here](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/ztp-metal3.md)
+
+In case the Bare Metal Operator on the HUB cluster version is 4.11 or higher, the integration between the MBO and the assisted-service is slightly different.
+The assisted-service will run a Preprovisioning Image Controller that reconcile the PreprovisioningImage for the  BareMetalHost.
```
Comment (avishayt) [inline docs/hive-integration/README.md:174]:
```suggestion
The assisted-service will run a Preprovisioning Image Controller that reconciles PreprovisioningImages and:
```

File: docs/hive-integration/README.md (hunk 1 above)
Comment (romfreiman) [reply docs/hive-integration/README.md:173]: Metal Bare Operator AKA: MBO

File: docs/hive-integration/README.md [hunk 4]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
...
+- Annotate the InfraEnv for the BareMetalHost with `infraenv.agent-install.openshift.io/enable-ironic-agent="true"`
+- Add the ironic agent config to the discovery ignition embeded to the InfraEnv ISO
+- Copy the InfraEnv ISODownloadURL to the PreprovisioningImage ImageUrl
+  This allows the host to register with metal3 and let metal3 reconcile the BareMetalHost hardware details.
```
Comment (avishayt) [inline docs/hive-integration/README.md:178]:
```suggestion
- Annotates the InfraEnv for the BareMetalHost with `infraenv.agent-install.openshift.io/enable-ironic-agent="true"`
- Adds the ironic agent config to the discovery ignition embedded in the InfraEnv ISO
- Copies the InfraEnv ISODownloadURL to the PreprovisioningImage ImageUrl, which allows the host to register with metal3 and let metal3 reconcile the BareMetalHost hardware details.
```

**Review by avishayt — COMMENTED** [review]
Nice, few small fixes

File: docs/hive-integration/README.md [hunk 5]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
...
+
+The Baremetal Agent Controller will will stop explicitly disabling *inspection* on `BareMetalHost`s it manages
```
Comment (romfreiman) [inline docs/hive-integration/README.md:180]: will will will

File: docs/hive-integration/README.md [hunk 6]
```diff
@@ -162,10 +162,37 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
...
+- Find the right pairs of BMH/Agent using their MAC addresses
+- Reconcile the Agent's spec by copying the following attributes from the BMH's annotations:
+  - Role: master/worker
+  - Hostname (optional for user to set)
+  - MachineConfigPool (optional for user to set)
+- Set the  BareMetalHost CustomDeploy Method to `start_assisted_install` - this custom deploy method will be added to start
+  the assisted agent when BMO requests deployment.
+
+The assisted agent will not reboot the machine at the end of the installation, instead it will stop
+the assisted agent service and let the ironic agent to manage the machine power state
```
Comment (romfreiman) [inline docs/hive-integration/README.md:192]: just to verify - ai will write all the coreos-install flags before the reboot, right? It's only the reboot that is missing

Comment (avishayt) [reply docs/hive-integration/README.md:192]: Yes

Comment (eranco74) [reply docs/hive-integration/README.md:192]: Right.
Also, note that the converged flow preserves the host installer args overrides and ignition config overrides functionality (annotations on the BMH are copied to the agent)

**Review by avishayt — APPROVED** [review]

File: docs/hive-integration/README.md [hunk 7]
```diff
@@ -162,10 +162,39 @@ In case that the Bare Metal Operator is installed, the Baremetal Agent Controlle
     - Hostname (optional for user to set)
     - MachineConfigPool (optional for user to set)
 - Reconcile the BareMetalHost hardware details by copying the Agent's inventory data to the BMH's `hardwaredetails` annotation.
+- Disable ironic inspection
 
 
 See BMAC documentation [here](./baremetal-agent-controller.md).
 
+## Bare Metal Operator Integration using the converged flow
```
Comment (romfreiman) [inline docs/hive-integration/README.md:170]: Lets document disks cleanup flow?

**Pull request merged**